trans_sec/controller/ddos_sdn_controller.py

```diff
@@ -45,14 +56,57 @@ def start(self, add_di):
         self.__main_loop()
 
     def stop(self):
         logger.info('Stopping SDN controller')
         self.running = False
         self.http_server.stop()
 
+    def create_drop_report(self):
+        for controller in self.controllers.values():
+            if controller == self.get_agg_controller():
+                logger.info("Creating drop report for controller: %s", controller)
+                match_keys, drop_count = controller.count_dropped_packets()
+                logger.info("Match keys - [%s]", match_keys)
+                logger.info('Dropped packet count - [%s]', drop_count)
+                if match_keys and drop_count:
+                    key_list = []
+                    for key, value in match_keys.items():
+                        key_list.append(value['value'])
+                    logger.info("Match key list - %s", key_list)
+                    keys = int(hashlib.md5(str(key_list).encode()).hexdigest(), 16)
+                else:
+                    keys = 0
+                    drop_count = 0
+                ip_len = IPV4_HDR_LEN + UDP_INT_HDR_LEN + DRPT_LEN + UDP_HDR_LEN + DRPT_PAYLOAD_LEN
+                udp_int_len = ip_len - IPV4_HDR_LEN
+                udp_len = UDP_HDR_LEN + DRPT_PAYLOAD_LEN
+                drop_pkt = Ether(type=consts.IPV4_TYPE)
+                drop_pkt = drop_pkt / IP(dst=self.topo['hosts']['ae']['public_ip'],
+                                         src=self.topo['hosts']['ae']['sdn_ip'], len=ip_len, proto=consts.UDP_PROTO)
+                drop_pkt = drop_pkt / UdpInt(sport=consts.UDP_INT_SRC_PORT,
+                                             dport=consts.UDP_TRPT_DST_PORT, len=udp_int_len)
+                drop_pkt = drop_pkt / DropReport(ver=consts.DRPT_VER,
+                                                 node_id=self.topo['switches']['aggregate']['int_id'],
+                                                 in_type=consts.DRPT_IN_TYPE,
+                                                 rpt_len=consts.DRPT_REP_LEN, md_len=consts.DRPT_MD_LEN,
+                                                 rep_md_bits=consts.DRPT_MD_BITS, domain_id=consts.TRPT_DOMAIN_ID,
+                                                 var_opt_bsmd=consts.DRPT_BS_MD,
+                                                 timestamp=int(time.time()),
+                                                 drop_count=drop_count,
+                                                 drop_tbl_keys=keys)
+                drop_pkt = drop_pkt / UDP(dport=consts.UDP_DRPT_DST_PORT, sport=consts.UDP_DRPT_SRC_PORT,
+                                          len=udp_len)
+                drop_pkt = drop_pkt / 'tps drop report'
+                try:
+                    sendp(drop_pkt, verbose=2)
+                    logger.info("Sent Drop Report packet")
+                except Exception as e:
+                    logger.info("Unable to send drop report - [%s]", e)
+        threading.Timer(10.0, self.create_drop_report).start()
```

Comment:
Can you instantiate the timer in the constructor and store it into a variable named something like:
self.drop_rpt_thread = threading.Timer(...)

Then in start line #59, replace with:
self.drop_rpt_thread.start()